health/health_json.c

```diff
@@ -231,6 +231,42 @@ static inline void health_rrdcalc2json_nolock(RRDHOST *host, BUFFER *wb, RRDCALC
 //
 //}
 
+void health_aggregate_alarms(RRDHOST *host, BUFFER *wb, BUFFER* contexts, RRDCALC_STATUS status) {
+    RRDCALC *rc;
+    int numberOfAlarms = 0;
+    char *tok = NULL;
+    char *p = NULL;
+
+    rrdhost_rdlock(host);
+
+    if (contexts) {
+        p = buffer_tostring(contexts);
+        while(p && *p && (tok = mystrsep(&p, ", |"))) {
+            if(!*tok) continue;
+
+            for(rc = host->alarms; rc ; rc = rc->next) {
+                if(unlikely(!rc->rrdset || !rc->rrdset->last_collected_time.tv_sec))
+                    continue;
+                if(unlikely(rc->rrdset && rc->rrdset->hash_context == simple_hash(tok)
+                            && !strcmp(rc->rrdset->context, tok) && rc->status == status))
```

Comment:
`rc->status == status`
should be replaced with
`status==RRDCALC_STATUS_RAISED?(rc->status >= RRDCALC_STATUS_WARNING):rc->status == status`